static/js/components/PersonalTab.js

```diff
@@ -23,15 +39,74 @@ class PersonalTab extends React.Component {
     ui:             UIState,
     nextStep:       () => void,
     prevStep:       () => void,
+    dispatch:       Dispatch,
+    dashboard:      DashboardState,
+    enrollments: ProgramEnrollmentsState,
+    currentProgramEnrollment:    ProgramEnrollment,
   };
 
+  programListing = (programs: Program) => {
+    const makeMenuItems = R.map(program => (
+      <MenuItem value={program.id} key={program.id} primaryText={program.title} />
+    ));
+    const sortPrograms = R.sortBy(R.compose(R.toLower, R.prop('title')));
+    const menuItems = R.compose(makeMenuItems, sortPrograms);
+    return menuItems(programs || {}) ;
+  }
+
+  setProgramHelper = (event: Event, key: string, value: string) => {
+    const {
+      dispatch,
+      enrollments: { programEnrollments },
+    } = this.props;
+    let selected = programEnrollments.find(enrollment => enrollment.id === value);
+    dispatch(setCurrentProgramEnrollment(selected));
```

Comment:
Instead of using `dispatch` in this file please make a function `setCurrentProgramEnrollment` in `ProfilePage` and pass it down as a prop